tests/test_project_script.py

```
@@ -158,3 +160,255 @@ def test_network_error_handled(self, mock_project_dir, capsys):
+        return self._tuple[key]
+
+
+class TestPythonVersionCheck:
+    """Tests for Python version checking in setup command."""
+
+    @pytest.fixture
+    def mock_project_dir(self, tmp_path):
+        """Create a temporary project directory structure."""
+        return tmp_path
+
+    def test_python_too_old_error(self, mock_project_dir, capsys):
```

Ruff (0.14.14)

[warning] Unused method argument: `mock_project_dir`

(ARG002)